### DISCUSSION TITLE
Migration of documentation to "tcmenu.github.io/documentation" and 4.3.1 release

### DISCUSSION THREAD

**Discussion by davetcc** [davetcc/tcMenu#512] [General]
We've migrated all of the documentation from thecoderscorner.com to github.io. Why may you ask? This is because it hosted privately on my website, and even though the library documentation was open, it was not possible for anyone else to build it and it fell under the same more restrictive terms as my website. It is now auto built on push (including all library reference docs) and hosted at https://tcmenu.github.io/documentation where I think you'll find everything you're used to.

Secondly, since we introduced color displays and the configurable renderer, themes have been a real bug bear for many, they were just too difficult to understand, and most didn't even try. A new front end on themes called `TcThemeBuilder` simplifies everything and makes it possible to easily either write a theme from scratch or adjust a theme yourself. The existing themes are still there, and you can move over at your own pace (should be truly little trouble to do so), the original ones are marked as deprecated.

Thirdly, we removed the checks against the online library-versions.xml from the UI and it now uses a GitHub public API to check the latest release instead. Given point 1, we can't easily put the library XML back.

**Comment by davetcc** [comment]
I'd like to thank @vzahradnik at this point, as he got the automated builds going that not only automatically tests every release, but also builds the website on push as well. These are all steps toward a lower maintenance and more sustainable TcMenu.

**Reply by vzahradnik** [reply]
Appreciated the credit, Dave :)
